src/zookeeperLock.ts

```diff
@@ -787,6 +792,39 @@ export class ZookeeperLock extends EventEmitter {
                                 });
                             }
                             debuglog(`${path}/${this.key}: lock not available for ${mySeq} on ${path}, waiting...`);
+
+                            let waitForIndex = 0;
+                            for (let i = 0; i < sequence.length; i++) {
+                                if (sequence[i].sequence === mySeq) {
+                                    waitForIndex = i - 1;
+                                    break;
+                                }
+                            }
+
+                            const retry = () => {
+                                if (this.continueLocking()) {
+                                    debuglog(`${path}/${this.key}: lock holder gone, retrying lock.`);
+                                    this.waitForLockHelper(resolve, reject, path);
+                                } else if (this.shouldRejectPromise()) {
+                                    return reject(new Error('aborting lock process'));
+                                }
+                            };
+
+                            debuglog('watching for key ' + `${path}/${sorted[waitForIndex].key}`);
+                            (this.client as any).exists(
```

Comment:
If I remember correctly, curator uses `getData` instead of `exists`. I wonder if it makes any difference here.